### ISSUE TITLE
Missing activity_id for MPI-ESM-1-2-HAM

### ISSUE THREAD

**Issue by MartinaSt** [WCRP-CMIP/CMIP6_CVs#937]
@durack1 This time only one missing activity_id registration was found. 

activity_id: ScenarioMIP is missing for 
source_id: MPI-ESM-1-2-HAM 
of institution_id: HAMMOZ-Consortium

**Comment by durack1** [comment]
@nedavid it appears that [`MPI-ESM-1-2-HAM` has published `ScenarioMIP` data on ESGF](https://esgf-node.llnl.gov/search/cmip6/?activity_id=ScenarioMIP&source_id=MPI-ESM-1-2-HAM), but this is not listed in the [activity participation field of the model registration](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/master/CMIP6_source_id.json#L5395-L5444).

Are there any other missing *MIPs/activity_id entries that we should also add for this? Or any other details that require amendment whilst this correction is implemented?

ping @MartinaSt

**Comment by nedavid** [comment]
The ssp370 experiment was published by mistake under ScenarioMIP. The experiment was run for AerChemMIP and should have been published under activity_id: AerChemMIP. This problem will be solved by re-publishing the data on ESGF under activity_id: AerChemMIP.

**Comment by durack1** [comment]
@nedavid thanks for the heads up, can you create an errata associated with the error please? Once generated it should list on https://errata.es-doc.org

**Comment by durack1** [comment]
@MartinaSt will close out this issue

**Comment by nedavid** [comment]
@durack1 please re-open this issue. I was now (re-)informed by DKRZ that the ssp370 experiment has to be published under ScenarioMIP, even if it is done for AerChemMIP. So please register MPI-ESM1.2-HAM for ScenarioMIP. Only this way a data citation for the ssp370 data of MPI-ESM1.2-HAM can be created apparently.

activity_id: ScenarioMIP
source_id: MPI-ESM-1-2-HAM
institution_id: HAMMOZ-Consortium

**Comment by nedavid** [comment]
see https://github.com/PCMDI/cmip6-cmor-tables/issues/264

**Comment by nedavid** [comment]
Please note that the HAMMOZ-Consortium will NOT do any other experiments for ScenarioMIP, regardless if they are required by ScenarioMIP or not. This registration is only requested to be able to publish the AerChemMIP data. Please let me know if this is not possible so I can inform DKRZ accordingly.

**Comment by taylor13** [comment]
We probably should have CMOR check whether the modeling group is registered for *any* of the activities responsible for an experiment and throw an error *only* if the group hasn't registered for any of them.  Currently, CMOR apparently requires the modeling group to register for the first one.  (I'll report an issue on cmor github.)
For now, though, I think the quickest solution is to simply register HAMMOZ-Consortium for ScenarioMIP.

**Comment by taylor13** [comment]
Before I request changes to CMOR, could you please clarify:  Has CMOR allowed the data to be published without the registering for ScenarioMIP?  Is the citation service the one that doesn't recognize that AerChemMIP is a co-sponsor of the experiment?

**Comment by wachsylon** [comment]
> We probably should have CMOR check whether the modeling group is registered for any of the activities responsible for an experiment and throw an error only if the group hasn't registered for any of them. Currently, CMOR apparently requires the modeling group to register for the first one. (I'll report an issue on cmor github.)

As I remember, CMOR/PrePARE does not test for any participation. 

>  Has CMOR allowed the data to be published without the registering for ScenarioMIP

Yes. `activity_participation` is not tested as we have discussed already in https://github.com/PCMDI/cmor/issues/434 . You said that

> as I recall, the activity_participation list was meant to be purely informal and not definitive.

from experience, I would confirm that. I do not think that this list is maintained in the CV by the modeling groups though it evolves with time when modeling group heads for a MIP change the institution and the MIP participation of the modeling group is lost. But I may be wrong...

If you instead aimed at asking whether CMOR allows `AerChemMIP` to be the `activity_id`  - no, definitely.

>  Is the citation service the one that doesn't recognize that AerChemMIP is a co-sponsor of the experiment?

Martina told me she would discuss the topic in the upcoming WIP telco. Maybe we can wait until then..

**Comment by MartinaSt** [comment]
Other modeling centers only publishing AerChemMIP just use AerChemMIP in activity_id for those experiments related to also ScenarioMIP. These datasets are not found in the ESGF searching for ScenarioMIP. There are also some published under ScenarioMIP not findable under AerChemMIP. A user has to specify the experiment_id to find all data published for this experiment_id: ssp370: 67081 datasets, ScenarioMIP and ssp370:  66421, AerChemMIP and ssp370: 61543.

The citation service does provide citations for the registered experiment data. The HAMMOZ-Consortium participates in AerChemMIP thus for all experiments belonging to this MIP, citations are provided. However, this data was not delivered as AerChemMIP but as ScenarioMIP data. Citations belong to data not to MIP design. It only requires that registration and data are consistent. 

The usual correction of datasets published under a wrong MIP was retract with errata information and publish under the correct MIP.

If we decide not to correct the error at its origin - in the data -, but decide for this CMOR-work-around, we create other problems like an artificial ScenarioMIP data citation with only ssp370 and a data citation for the AerChemMIP contribution without ssp370.

**Comment by MartinaSt** [comment]
Just to clarify the suggested data correction: activity_drs=AerChemMIP

**Comment by wachsylon** [comment]
> If we decide not to correct the error at its origin - in the data -, but decide for this CMOR-work-around,

For me, the CMOR changes allows us to change the data so it is closely connected.

> activity_drs=AerChemMIP
> activity_id= AerChemMIP ScenarioMIP

I think that if we allow that, it would be a solution.
@MartinaSt 
We would need to make sure that there are no AerChemMIP participants that already published to ScenarioMIP thinking that this is the only solution to publish ssp370. If they now realize there is another method, we may be busy with retracting and republishing lots of files... The number of experiments that share activities is 12. Besides, consistency...

Also, there are modeling groups submitting data for both AerChemMIP and ScenarioMIP. Do they get citation for both with just one activity_id?

**Comment by taylor13** [comment]
The original intention was to ensure that *all* of the ssp370 data would be found under the ScenarioMIP directory constructed according to the DRS defined in  [the global attributes, filenames, etc. document](https://goo.gl/v1drZl):
```
Directory structure = <mip_era>/
                                 <activity_id>/
                                          <institution_id>/
                                                     <source_id>/.........
```
If this DRS directory structure is always followed, then a computer code could read in ssp370 experiment data from the *same* activity_id subdirectory: "ScenarioMIP".  If we allow "AerChemMIP" as an option, a code has to be much smarter and look in both the "ScenarioMIP" and AerChemMIP"  directories to get data from all the models that performed the ssp370 experiment.  
I would like everyone to follow the DRS as it was intended.  To do this, everyone should list both activities in the activity_id global attribute, and the SenarioMIP should appear first as it does in the CMIP6_experiment_id.CV.
I don't know if we should bother cleaning up the data that was originally written to the wrong directory structure, but new datasets should adhere to these rules.
What are the consequences for the citation service?

**Comment by sjmarsland** [comment]
Thanks Karl,

"everyone should list both activities in the activity_id global attribute”

Is that everyone in AerChemMIP should register for ScenarioMIP?

Or every model in ScenarioMIP with an ssp-370 should also register AirChemMIP as an activity_id?

On 17 Jun 2020, at 12:45 am, taylor13 <notifications@github.com<mailto:notifications@github.com>> wrote:


The original intention was to ensure that all of the ssp370 data would be found under the ScenarioMIP directory constructed according to the DRS defined in the global attributes, filenames, etc. document:

Directory structure = <mip_era>/
                                 <activity_id>/
                                          <institution_id>/
                                                     <source_id>/.........

If this DRS directory structure is always followed, then a computer code could read in ssp370 experiment data from the same activity_id subdirectory: "ScenarioMIP". If we allow "AerChemMIP" as an option, a code has to be much smarter and look in both the "ScenarioMIP" and AerChemMIP" directories to get data from all the models that performed the ssp370 experiment.
I would like everyone to follow the DRS as it was intended. To do this, everyone should list both activities in the activity_id global attribute, and the SenarioMIP should appear first as it does in the CMIP6_experiment_id.CV.
I don't know if we should bother cleaning up the data that was originally written to the wrong directory structure, but new datasets should adhere to these rules.
What are the consequences for the citation service?

**Comment by MartinaSt** [comment]
@wachsylon Great that this might be a solution.

The data for those who participate in both - AerChemMIP and ScenarioMIP - the data exist only once and so does the citation. That is the point, why the citation is linked to the data and not to the design. Participants either specify the contribution to AerChemMIP or ScenarioMIP in their datasets. 

@taylor13 Yes, the implementation of the ESGF then ensures that the data can be found under both activity_ids but the data exist once under a specific DRS and a specific dataset_id and with one specific  citation based on the DRS. So far, modeling centers with publications under a wrong MIP agreed to correct their data. The question is, do we need an exception for this data from the HAMMOZ-Consortium?

If we let old dataset versions remain under the wrong and we have new datasets under the right DRS, then we would have 2 references on the 2 citation granularities for the subset under the correct DRS in case of non-registration of ScenarioMIP and in case of registration we would have 4 Citations - 2 for the contributions to the different MIPs per model and 2 for the contributions to the experiments contributed to the two different MIPs.

**Comment by wachsylon** [comment]
There are activites like LS3MIP requesting data from ssp370 with tier one priority. Why is LS3MIP not listed as `activity_id` for that experiment? Or, in another way, why is AerChemMIP listed at all then?

Either way, full contribution to one MIP meaning having citation for all tier1 experiments for a MIP will not be possible :

> The data for those who participate in both - AerChemMIP and ScenarioMIP - the data exist only once and so does the citation.

**Comment by durack1** [comment]
@nedavid that tweak is now in place, please take a peek at [CMIP6_source_id.html](https://wcrp-cmip.github.io/CMIP6_CVs/docs/CMIP6_source_id.html) and let me know of any further tweaks required.

@taylor13 @MartinaSt @wachsylon it is not clear to me if further discussion needs to be moved out of this issue, to a new issue, or whether we can close this out. I haven't waiting on feedback

**Comment by MartinaSt** [comment]
I will wait before I include the additional MIP in the citation until the discussion is settled, as Fabian's  current idea breaks the citation.

**Comment by wachsylon** [comment]
> Fabian's current idea breaks the citation.

So it seems to me there is no final decision on how to proceed. 

If ssp370 should only be published under ScenarioMIP and, if, in addition, the double entries in `activity_id` suggest that it could also be possible to publish it under AerChemMIP - then let us just remove all second entries of double entries in activtiy_id of experiments.

**Comment by martinjuckes** [comment]
As far as I can see, it should not be necessary for a model to say they are participating in a MIP just because they are producing data from and experiment defined by that MIP. If a model is participating in `AerChemMIP`, for instance, then they should provide the data requested in `AerChemMIP` ... which might include data from experiments defined in other MIPs. The can't mean that the model then has to participate in those MIPs and produce all the data requested by them. 

In the example we have here, the `MPI-ESM-1-2-HAM` model is only providing data from one `ScenarioMIP` experiment, so it is far short of the contribution expected from models which are actually participating in `ScenarioMIP` (unless we introduce a new category of partial contribution).

If the ESGF `Activity` tab is meant to select models based on their participation, then having them excluded from `ScenarioMIP` looks like the correct answer to me.  The data which they have submitted is from the `AerChemMIP` request .. so people should, under the current configuration, look for it under `AerChemMIP`. 

If you want to list all data in experiments defined by `ScenarioMIP` you would need a different means of selecting ... but you can't have both the current functionality and that functionality in a single button.

**Comment by sol1105** [comment]
There are other MIPs requesting data from the ssp370 experiment as well (VIACSAB, C4MIP, LUMIP, GeoMIP, DynVarMIP, SIMIP, LS3MIP as well as the default CMIP-request). These MIPs are not listed as activities of ssp370. 
So should it also be allowed to publish ssp370-data via activity C4MIP etc?
And if not, why should it then be allowed to select between ScenarioMIP and AerChemMIP, refering to the data request?
I think it would complicate things to allow the selection of an activity based on the data request.
According to the CMIP6 DRS document the first mentioned activity is selected for the DRS-path. Afaik newer versions of PrePARE are checking this.
Introducing a "partial participation" for CMIP6 might be a good compromise and the way to go, as the source_id-CV was interpreted in two different ways:
On the one hand, the citation service uses the information given in the source_id CV as if it documents the official activity registration. On the other hand the activity registration information is regarded as informal (https://github.com/PCMDI/cmor/issues/434) and also PrePARE does not check it.

**Comment by durack1** [comment]
Just for context, in the [`experiment_id` CV of the CMIP6_CVs](https://wcrp-cmip.github.io/CMIP6_CVs/docs/CMIP6_experiment_id.html), the number of experiments that list more than a single activity include:

|`experiment_id`|`activity_id` (primary)|`activity_id` (additional)|
|--|--|--|
|dcppC-forecast-addPinatubo|DCPP|VolMIP|
|esm-1pct-brch-1000PgC|C4MIP|CDRMIP|
|esm-1pct-brch-2000PgC|C4MIP|CDRMIP|
|esm-1pct-brch-750PgC|C4MIP|CDRMIP|
|esm-1pctCO2|C4MIP|CDRMIP|
|esm-bell-1000PgC|C4MIP|CDRMIP|
|esm-bell-2000PgC|C4MIP|CDRMIP|
|esm-bell-750PgC|C4MIP|CDRMIP|
|land-hist|LS3MIP|LUMIP|
|piClim-aer|RFMIP|AerChemMIP|
|piClim-control|RFMIP|AerChemMIP|
|ssp370|ScenarioMIP|AerChemMIP|

**Comment by martinjuckes** [comment]
Hi @sol1105 : what do you mean "publish ssp370 data via C4MIP"? `ssp370` is a ScenarioMIP experiment -- seems simple enough.

**Comment by durack1** [comment]
Closing this issue as discussion should rather be directed into #951
